mne/viz/ica.py

```diff
@@ -329,17 +335,42 @@ def plot_ica_properties(ica, inst, picks=None, axes=None, dB=True,
 
     # calculations
     # ------------
+    
     if isinstance(inst, BaseRaw):
+        # if reject is set, reject
+        if reject == 'auto':
+            reject = getattr(ica, 'reject_', None)
+        if reject is not None:
+            data = inst.get_data(ica.picks_)
+            data, drop_inds = _reject_data_segments(data, reject, flat=None, 
+                                                    decim=None, info=ica.info,
+                                                    tstep=2.0)
+            ch_names = np.array(inst.info['ch_names'])
+            ch_names = ch_names[ica.picks_].tolist()
+            ch_type = ica.info['chs']
+            # TODO : check if we need to create the info
+            info = create_info(ch_names=ch_names, sfreq=ica.info['sfreq'])
+            inst = RawArray(data, ica.info)
+        else:
+            drop_inds = None
         # break up continuous signal into segments
         from ..epochs import _segment_raw
         inst = _segment_raw(inst, segment_length=2., verbose=False,
                             preload=True)
         kind = "Segment"
     else:
         kind = "Epochs"
-
+    
     epochs_src = ica.get_sources(inst)
-    ica_data = np.swapaxes(epochs_src.get_data()[:, picks, :], 0, 1)
+
+    data = epochs_src.get_data()
+
+    ica_data = np.swapaxes(data[:, picks, :], 0, 1)  
+
+    # getting dropped epochs indexes
+    dropped_indexes = []
+    for dropped in drop_inds:
+        dropped_indexes.append(int(dropped[0]/len(inst)))
```

Comment:
You could use epochs.selection (invert it via bool mask) to avoid computation and looping if I'm not mistaken.

Reply:
Ok forget what I said, I did not read properly. These two lines till look intuitively suspicious to me. To be deep-checked.

Reply:
Yes, I was not sure about these yesterday, I will double check today :)